finagle-thrift/src/main/scala/com/twitter/finagle/Thrift.scala

```diff
@@ -115,18 +124,9 @@ object Thrift extends Client[ThriftClientRequest, Array[Byte]] with ThriftRichCl
           val Label(label) = params[Label]
           val param.ClientId(clientId) = params[param.ClientId]
           val param.ProtocolFactory(pf) = params[param.ProtocolFactory]
-          val Stats(stats) = params[Stats]
-          val preparer = new ThriftClientPreparer(pf, label, clientId)
-          val underlying = preparer.prepare(next, params)
-          new ServiceFactoryProxy(underlying) {
-            val stat = stats.stat("codec_connection_preparation_latency_ms")
-            override def apply(conn: ClientConnection) = {
-              val elapsed = Stopwatch.start()
-              super.apply(conn).ensure {
-                stat.add(elapsed().inMilliseconds)
-              }
-            }
-          }
+          val param.AttemptProtocolUpgrade(upgrade) = params[param.AttemptProtocolUpgrade]
```

Comment:
Why we need to extract `AttemptProtocolUpgrade` from params and pass it separately? Preparer already takes `params` so you can extract them later.